src/MeshKernel/CurvilinearGridDeRefinement.cpp

```diff
@@ -87,6 +77,7 @@ CurvilinearGrid CurvilinearGridDeRefinement::Compute()
         }
         mIndexOriginalGrid += localMDeRefinement;
     }
+
     // substitute original grid with the derefined one
-    return CurvilinearGrid(std::move(deRefinedGrid), m_grid->m_projection);
-}
+    return std::make_shared<CurvilinearGrid>(std::move(deRefinedGrid), m_grid->m_projection);
```

Comment:
be aware that because you use shared pointers everywhere, some instances could still keep the old curvilinear grid alive. maybe thats what you want, maybe not